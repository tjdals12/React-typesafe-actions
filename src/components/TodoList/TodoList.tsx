import React from 'react';
import classNames from 'classnames/bind';
import TodoItem from 'components/TodoItem';
import useTodos from 'hooks/useTodos';
import styles from './TodoList.scss';

const cx = classNames.bind(styles);

export default function TodoList(): JSX.Element {
    const todos = useTodos();

    return (
        <div className={cx('todo-list')}>
            {todos.map(todo => (
                <TodoItem key={todo.id} todo={todo} />
            ))}
        </div>
    );
}
