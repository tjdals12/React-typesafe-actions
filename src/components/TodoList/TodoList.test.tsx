import React from 'react';
import { render } from '@testing-library/react';
import TodoList from './TodoList';

describe('<TodoList />', () => {
    it('matches snapshot', () => {
        const utils = render(<TodoList />);
        expect(utils.container).toMatchSnapshot();
    });
});
