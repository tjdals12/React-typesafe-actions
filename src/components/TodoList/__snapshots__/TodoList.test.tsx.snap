// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TodoList /> matches snapshot 1`] = `
<div>
  <div
    class="todo-list"
  >
    <li
      class="todo-item"
    >
      <input
        checked=""
        class="input done"
        readonly=""
        type="checkbox"
      />
      <span
        class="text"
      >
        Something 1
      </span>
      <span
        class="remove"
      >
        X
      </span>
    </li>
    <li
      class="todo-item"
    >
      <input
        class="input "
        readonly=""
        type="checkbox"
      />
      <span
        class="text"
      >
        Something 2
      </span>
      <span
        class="remove"
      >
        X
      </span>
    </li>
    <li
      class="todo-item"
    >
      <input
        class="input "
        readonly=""
        type="checkbox"
      />
      <span
        class="text"
      >
        Something 3
      </span>
      <span
        class="remove"
      >
        X
      </span>
    </li>
  </div>
</div>
`;
